### DISCUSSION TITLE
Want to know how to enum structure blocks

### DISCUSSION THREAD

**Discussion by ArcadeCode** [maruohon/litematica#801] [Q&A] [answered]
Hello, I have a project of calculator in minecraft to get a list of block of a structure and give user all basics blocks to get, like if the structure use Furnaces, we need a certain amount of cobblestones.

But for this I checked how nbt work and It was such complex so I search If anybody have an tool for this. With the list of material in Litematica I thinks the project have what I want, so anybody can show me what's script read structures and how list materials ?

**Comment by maruohon** [comment, marked as answer]
Do you need this in a mod, or in an external script?

If you are making a mod, then you can directly use Litematica's `MaterialList*` classes to get the list of `ItemStack`s and the total number of items per stack for a given schematic.

If you need it in an external script, then there are a number of libraries people have made for various languages for accessing Litematica schematics. I know of libraries for Python, Rust and JavaScript.

**Reply by ArcadeCode** [reply]
Thank you so much for your help !

**Reply by maruohon** [reply]
Did that already solve the problem for you? Then I assume you were working on a mod, since I didn't even link any of the external libraries yet.

**Reply by ArcadeCode** [reply]
Oh yeah I made a mod but if you have librairies for this I take.

**Reply by maruohon** [reply]
The Litematica schematic format is described to some extent here:
https://github.com/maruohon/litematica/issues/53

Also some more stuff on my Discord, for example starting from these messages:
- https://discord.com/channels/169369095538606080/566649314001158165/1176569283271213098
- https://discord.com/channels/169369095538606080/566649314001158165/1097540707931127938
- https://discord.com/channels/169369095538606080/169369095538606080/1081089742944337941
- https://discord.com/channels/169369095538606080/566649314001158165/1097564948999184394

Some libraries or other schematic related code:
- Python: https://pypi.org/project/litemapy/
- Python: https://github.com/Kikugie/pymatic
- Rust: https://crates.io/crates/rustmatica
- Julia: https://github.com/lntricate1/Litematica.jl
- JavaScript: https://github.com/mms0316/mms0316.github.io/blob/master/js/schematics.js

**Reply by ArcadeCode** [reply]
Waouh that's a lot of ressources, thank for your investissement with this I think I earn a lot of time 😁
